clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1034,38 +1049,11 @@ private void getConfigKeyRst(ConfigKey def, StringBuilder b) {
     }
 
     /**
-     * Get a list of configs sorted into "natural" order: listing required fields first, then
-     * ordering by importance, and finally by name.
-     */
-    protected List<ConfigKey> sortedConfigs() {
-        // sort first required fields, then by importance, then name
-        List<ConfigKey> configs = new ArrayList<>(this.configKeys.values());
-        Collections.sort(configs, new Comparator<ConfigKey>() {
-            public int compare(ConfigKey k1, ConfigKey k2) {
-                // first take anything with no default value
-                if (!k1.hasDefault() && k2.hasDefault()) {
-                    return -1;
-                } else if (!k2.hasDefault() && k1.hasDefault()) {
-                    return 1;
-                }
-
-                // then sort by importance
-                int cmp = k1.importance.compareTo(k2.importance);
-                if (cmp == 0) {
-                    // then sort in alphabetical order
-                    return k1.name.compareTo(k2.name);
-                } else {
-                    return cmp;
-                }
-            }
-        });
-        return configs;
-    }
-
-    /**
      * Get a list of configs sorted taking the 'group' and 'orderInGroup' into account.
+     *
+     * If grouping is not specified, the result will reflect "natural" order: listing required fields first, then ordering by importance, and finally by name.
      */
-    protected List<ConfigKey> sortedConfigsByGroup() {
+    public List<ConfigKey> sortedConfigs() {
```

Comment:
I think I should be able to avoid it by doing this bit from within `ConfigDef`. I'll try it

```
for (ConfigDef.ConfigKey key : transformationConfigDef.sortedConfigs()) {
                newDef.define(new ConfigDef.ConfigKey(
                        prefix + key.name,
                        key.type,
                        key.defaultValue,
                        TransformationConfig.embeddedValidator(prefix, key.validator),
                        key.importance,
                        key.documentation,
                        groupPrefix + (key.group == null ? "" : ": " + key.group),
                        orderInGroup++,
                        key.width,
                        key.displayName,
                        TransformationConfig.embeddedDependents(prefix, key.dependents),
                        TransformationConfig.embeddedRecommender(prefix, key.recommender)
                ));
            }
```